scripts/pull_from_upstream.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -eux
+
+cd $1
+
+TOOLCHAIN_DATE=$2
+COMMIT_HASH=$3
+
+# The checkout and pull itself needs to happen in sync with features/verify-rust-std
+# Often times rust is going to be ahead of kani in terms of the toolchain, and both need to point to
+# the same commit
+SYNC_BRANCH="sync-$TOOLCHAIN_DATE" && echo "--- Fork branch: ${SYNC_BRANCH} ---"
+# # 1. Update the upstream/master branch with the latest changes
+git fetch upstream
+git checkout $COMMIT_HASH
+
+# # 2. Update the subtree branch
+git subtree split --prefix=library --onto subtree/library -b subtree/library
```

Comment:
Could you try to add `--rejoin` to see whether this can give us much better performance? Git's documentation suggests that git will try to optimise so as to only add new commits. (And I think you wouldn't need the `subtree merge` thereafter.)

Reply:
Offline discussion suggests that this doesn't actually help.